spark/src/test/scala/org/apache/comet/DataGenerator.scala

```diff
@@ -95,4 +102,38 @@ class DataGenerator(r: Random) {
       Range(0, n).map(_ => r.nextLong())
   }
 
+  // Generate a random row according to the schema, the string filed in the struct could be
+  // configured to generate strings by passing a stringGen function. Other types are delegated
+  // to Spark's RandomDataGenerator.
+  def generateRow(schema: StructType, stringGen: Option[() => String] = None): Row = {
+    val fields = mutable.ArrayBuffer.empty[Any]
+    schema.fields.foreach { f =>
+      f.dataType match {
+        case StructType(children) =>
+          fields += generateRow(StructType(children), stringGen)
```

Comment:
This could be a `map` operation to remove the mutable buffer and simplify the code slightly. For example:

```rust
    val fields = schema.fields.map { f =>
      f.dataType match {
        case StructType(children) =>
          generateRow(StructType(children), stringGen)
        case ...
```